htdocs/core/class/html.form.class.php

```diff
@@ -1872,6 +1872,7 @@ function select_produits_list($selected='',$htmlname='productid',$filtertype='',
             $selectFields.= ", price_rowid, price_by_qty";
 		}
         $sql.= " FROM ".MAIN_DB_PREFIX."product as p";
+	$sql.= " LEFT JOIN ".MAIN_DB_PREFIX."product_fournisseur_price as pfp ON p.rowid = pfp.fk_product";
```

Comment:
This introduce errors in calculation of sum(ps.reel) when filter on stock status is used.
Solution. Search for supplier ref to get product ids that match the ref, dans add a where on rowid in (product_ids, ...).